src/log_batch.rs

```diff
@@ -90,14 +90,29 @@ impl EntryIndexes {
         Ok(Self(entry_indexes))
     }
 
+    #[allow(unused_mut)]
     pub fn encode(&self, buf: &mut Vec<u8>) -> Result<()> {
-        let count = self.0.len() as u64;
+        let mut count = self.0.len() as u64;
+        #[cfg(feature = "failpoints")]
+        {
+            let enable_hole = || {
+                fail::fail_point!("memtable::enable_hole", |_| true);
+                false
+            };
+
+            if enable_hole() {
+                count -= 1;
+            }
+        }
```

Comment:
Why still add a new failpoint?

Reply:
You mean we can use the name `corrupted_items` instead of create a new name ? or we don't need to add a failpoint at all and can add some logic in the failpoint `corrupted_items` ?

Reply:
I mean why the failpoint "corrupted_items" isn't enough for your use case?

Reply:
On the one size `corrupted_items`  is intended for physical corruption and I think logical hole is different from that.  On the other hand. It's hard to modify byte array to create a logical hole as all content has been serialized to byte array (We may need to move the byte array content to remove the item that you want to delete) So i add a new failedpoint to mock the scenario that the end index of each `EntryIndexes` is not continuous.

Correct me if i'm wrong, thanks

Reply:
You can only write one log entry in one log batch, and corrupt that log batch. Also, corruption of multiple log entries (inside one log batch) still counts as a logical hole, I don't see why you can't test it.